### PR TITLE
Better decoding of sender and subject in mailq.php #630

### LINKED ISSUE #630: Wrong character display in mailq.php
I've found this.

When I display the Exim mailq in GUI, some Subject: are not well printed.

Look at the word `série` and `Frontière`. The `é` is replaced by a `?`.

capture d ecran 2017-03-10 a 18 50 04

But in detail.php, it's ok for the same ID.

capture d ecran 2017-03-10 a 18 50 49

But not all Subject: are not well displayed in mailq.php. Some are ok.

capture d ecran 2017-03-10 a 19 01 33

Crazy!

### PR DESCRIPTION
**Pull request by stefaweb**
- Add imap_mime_header_decode() when available (better decoding)
- Corrected a problem with `Subject:` variable that was cut (no multiline header mgt)
- Added multiline mgt for `From:`
- Print sender in GUI if envelopesender is empty (bounce)
- Initialized sender and subject variables to ovoid warning
- Added new mime decoder for `Subject:`
- Removed unused code
- Corrected bug for Message-ID
- Add getFROMheader($header) in functions.php
- Add getSUBJECTheader($header) in functions.php
- Added MAIL_SENDER in conf.php
- Added STATUS_REFRESH in mails.php

### CODE REVIEW

File: tools/Sendmail-Exim_queue/mailwatch_sendmail_queue.php [hunk 1]
```diff
@@ -247,6 +247,8 @@
         if (!empty($output)) {
             foreach ($output as $msgid => $msginfo) {
                 // Insert each record
+                $sender = preg_replace('~[\r\n\t]+~', '', getUTF8String($msginfo['sender']));
```
Comment (Skywalker-11) [inline tools/Sendmail-Exim_queue/mailwatch_sendmail_queue.php]: You should check if sender and subject are there with isset

Comment (stefaweb) [inline tools/Sendmail-Exim_queue/mailwatch_sendmail_queue.php]: How I do that cleanly?

Comment (Skywalker-11) [inline tools/Sendmail-Exim_queue/mailwatch_sendmail_queue.php]:
```
$sender = isset($msginfo['sender']) ? $msginfo['sender'] : "";
$sender = preg_replace('~[\r\n\t]+~', '', getUTF8String($sender);
```
or 
```
$sender = preg_replace('~[\r\n\t]+~', '', getUTF8String(isset($msginfo['sender']) ? $msginfo['sender'] : "");
```

**Comment by Skywalker-11** [comment]
In case sender or subject are not set for the mail PHP would throw an error or notice if we don't check it.

File: tools/Sendmail-Exim_queue/mailwatch_sendmail_queue.php (hunk 1 above)
Comment (stefaweb) [inline tools/Sendmail-Exim_queue/mailwatch_sendmail_queue.php]: one `)` missing.

$sender = preg_replace('~[\r\n\t]+~', '', getUTF8String(isset($msginfo['sender'])) ? $msginfo['sender'] : "");

**Comment by stefaweb** [comment]
Sorry, commented = line 113

**Comment by Skywalker-11** [comment]
Its there from the first version of the file so the case can be deleted

**Comment by stefaweb** [comment]
Subject "cut" is now working.

Still problem with characters.

**Comment by stefaweb** [comment]
Can be merged.

**Comment by stefaweb** [comment]
In function.


```
/**
 * @param $header
 * @return string
 */
function getFROMheader($header)
{
        $sender = "";
	if (preg_match('/^\d{3}F From: (.*)(\n\s+(.*))*/im', $header, $match)) {
	    if (isset($match[1])) {
	        $sender = $match[1];
	    }
	    if (isset($match[2])) {
	        $sender .= " " . $match[2];
	    }
	    if (isset($match[3])) {
	        $sender .= " " . $match[3];
	    }
	    if (preg_match('/\S+@\S+/', $sender, $match_email)) {
	        if (isset($match_email[0])) {
	            $sender = str_replace(array('<', '>', '"'), '', $match_email[0]);
	        }
	    }
	}
    return $sender;
}

```

**Comment by stefaweb** [comment]
Ready to merge.

**Comment by stefaweb** [comment]
Still two problems to solve.

- Subject without space.

```
128 Subject: [la ptite broc 
sans la grosse 
tête 27 28 78] 
Solderie bazar 
Dreux
```

- sometime I received this. One error all 2 hours for 1000 mails per hour.
`Notice: iconv(): Detected an incomplete multibyte character in input string in /var/www/html/mailscanner/functions.php on line 1032`

In this line.

`$convLine .= iconv($linePartArr[$j]->charset, 'UTF-8', $linePartArr[$j]->text);
`

**Comment by stefaweb** [comment]
Last problem solved.

ok for me.

**Comment by stefaweb** [comment]
@endelwar 

Please merge this one. I need it.

**Pull request merged**